gke/autopilot/labels/main.tf

```diff
@@ -0,0 +1,34 @@
+/**
+* Copyright 2024 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+# [START gke_autopilot_labels]
+resource "google_container_cluster" "default" {
+  name     = "gke-autopilot-labels"
+  location = "us-central1"
+
+  enable_autopilot = true
+
+  node_config {
+    labels = {
```

Comment:
autopilot can have cluster labels, could it have nodepool labels?

Reply:
Hmm - Good point, you can do a `cluster describe` and see the AutoPilot labels, so I suspect they will appear, however that wasn't the actual goal.  I'll manually test that works, and then revise/clarify this example for both cluster `resource_labels` and nodePool `node__config.labels` and circle back.  Thanks @yliaog!

Reply:
Turns out `google_container_node_pool` doesn't include an option for `resource_labels`.